Generated di/decorators.d.ts missing annotations after di refactor

Here's the log when using 2.0.0-alpha.31 from npm:
```log
node_modules/angular2/src/di/decorators.d.ts(34,13): error TS1003: Identifier expected.
node_modules/angular2/src/di/decorators.d.ts(37,17): error TS1003: Identifier expected.
node_modules/angular2/src/di/decorators.d.ts(45,13): error TS1003: Identifier expected.
node_modules/angular2/src/di/decorators.d.ts(48,17): error TS1003: Identifier expected.
node_modules/angular2/src/di/decorators.d.ts(56,13): error TS1003: Identifier expected.
node_modules/angular2/src/di/decorators.d.ts(59,17): error TS1003: Identifier expected.
```

And here's what those lines look like:

```typescript
export interface ParentFactory {
    ({self: }?: {
        self: any;
    }): any;
    new ({self: }?: {
        self: any;
    }): ParentMetadata;
}
/**
 * Factory for creating {@link AncestorMetadata}.
 */
export interface AncestorFactory {
    ({self: }?: {
        self: any;
    }): any;
    new ({self: }?: {
        self: any;
    }): AncestorMetadata;
}
/**
 * Factory for creating {@link UnboundedMetadata}.
 */
export interface UnboundedFactory {
    ({self: }?: {
        self: any;
    }): any;
    new ({self: }?: {
        self: any;
    }): UnboundedMetadata;
}
```

--- a/modules/angular2/src/di/decorators.ts
+++ b/modules/angular2/src/di/decorators.ts
@@ -46,24 +46,24 @@ export interface SelfFactory {
  * Factory for creating {@link ParentMetadata}.
  */
 export interface ParentFactory {
-  ({self: boolean}?): any;
-  new ({self: boolean}?): ParentMetadata;
+  (visibility?: {self: boolean}): any;
+  new (visibility?: {self: boolean}): ParentMetadata;
 }
 
 /**
  * Factory for creating {@link AncestorMetadata}.
  */
 export interface AncestorFactory {
-  ({self: boolean}?): any;
-  new ({self: boolean}?): AncestorMetadata;
+  (visibility?: {self: boolean}): any;
+  new (visibility?: {self: boolean}): AncestorMetadata;
 }
 
 /**
  * Factory for creating {@link UnboundedMetadata}.
  */
 export interface UnboundedFactory {
-  ({self: boolean}?): any;
-  new ({self: boolean}?): UnboundedMetadata;
+  (visibility?: {self: boolean}): any;
+  new (visibility?: {self: boolean}): UnboundedMetadata;
 }
 
 /**

